pkg/controller/core/clusterqueue_controller.go

```diff
@@ -497,3 +552,67 @@ func (r *ClusterQueueReconciler) updateCqStatusIfChanged(
 	}
 	return nil
 }
+
+func (r *ClusterQueueReconciler) getWorkloadsStatus(cq *kueue.ClusterQueue) *kueue.ClusterQueuePendingWorkloadsStatus {
```

Comment:
I think this function should set `nil` if the feature is disabled. I suggest to introduce a helper function `isVisibilityEnabled()` that would check the `MaxCount=0`. Then, use this function here, and in `Start`.

Reply:
function was added https://github.com/kubernetes-sigs/kueue/pull/1069/commits/3b30bd48c7e9b9de73d3d87d00bd3830281a8819